examples/connect/shiny-python/app.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# mypy: ignore-errors
+import os
+
+import pandas as pd
+
+from posit.connect.external.databricks import viewer_credentials_provider
+
+from databricks import sql
+from databricks.sdk.service.iam import CurrentUserAPI
+from databricks.sdk.core import ApiClient, Config
+
+from shiny import App, Inputs, Outputs, Session, render, ui
+
+DATABRICKS_PAT = os.getenv("DATABRICKS_TOKEN")
+DATABRICKS_HOST=os.getenv("DATABRICKS_HOST")
+DATABRICKS_HOST_URL = f"https://{DATABRICKS_HOST}"
+SQL_HTTP_PATH = os.getenv("DATABRICKS_PATH")
+
+app_ui = ui.page_fluid(
+    ui.output_text("text"),
+    ui.output_data_frame("result")
+)
+
+def get_connection_settings(session_token: str):
+    """
+    Construct a connection settings dictionary that works locally and on Posit Connect.
+    """
+
+    if os.getenv("CONNECT_SERVER"):
+        return {
+            "server_hostname": DATABRICKS_HOST,
+            "http_path": SQL_HTTP_PATH,
+            "auth_type": "databricks-oauth",
+            "credentials_provider": viewer_credentials_provider(user_session_token=session_token)
+        }
+    return {
+        "server_hostname": DATABRICKS_HOST,
+        "http_path": SQL_HTTP_PATH,
+        "access_token": DATABRICKS_PAT
+    }
+
+def get_databricks_user_info(session_token: str):
+    """
+    Use the Databricks SDK to get the current user's information.
+    """
+
+    if os.getenv("CONNECT_SERVER"):
+        cfg = Config(
+            host=DATABRICKS_HOST_URL,
+            credentials_provider=viewer_credentials_provider(user_session_token=session_token)
+        )
+    else:
+        cfg = Config(host=DATABRICKS_HOST_URL, token=DATABRICKS_PAT)
+
+    return CurrentUserAPI(ApiClient(cfg)).me()
+
+def server(input: Inputs, output: Outputs, session: Session):
+    """
+    Shiny for Python example application that shows user information and 
+    the first few rows from a table hosted in Databricks.
+    """
+
+    session_token = session.http_conn.headers.get('Posit-Connect-User-Session-Token')
+    databricks_user_info = get_databricks_user_info(session_token=session_token)
+
+    @render.data_frame
+    def result():
+
+        query = "SELECT * FROM samples.nyctaxi.trips LIMIT 10;"
+        connection_settings = get_connection_settings(session_token=session_token)
+
+        with sql.connect(**connection_settings) as connection:
```

Comment:
Similarly, these connection settings should be valid for both local execution and when running on Connect:

```python
with sql.connect( 
    ... 
    auth_type="databricks-oauth", 
    credentials_provider=credentials_provider):
```

The [streamlit example](https://github.com/posit-dev/posit-sdk-py/blob/fe0e95d39f4c884f9abc4db063eb4ca65a57c283/examples/connect/streamlit/sample-content.py#L37-L42) does this as well, without the PAT

This works because of the value `auth_type="databricks-oauth"`. This tells the databricks sql sdk to do local oauth `if credentials_provider == None`

https://github.com/databricks/databricks-sql-python/blob/f6fd7a7956a4dbc78ad36b5e079fe8d74176a0f1/src/databricks/sql/auth/auth.py#L51-L65

Reply:
Misunderstood how this works. SDK FTW! Removed all unnecessary code